src/Bicep.Cli.IntegrationTests/RestoreCommandTests.cs

```diff
@@ -257,10 +258,9 @@ public async Task Restore_Artifacts_BackwardsAndForwardsCompatibility(string? me
         [DataRow(new string[] { "unknown1", "unknown2" }, "Did not expect only layer media types unknown1, unknown2")]
         [DataRow(new string[] { BicepMediaTypes.BicepModuleLayerV1Json, BicepMediaTypes.BicepModuleLayerV1Json },
             $"Did not expect to find multiple layer media types of application/vnd.ms.bicep.module.layer.v1\\+json, application/vnd.ms.bicep.module.layer.v1\\+json")]
-        [DataRow(new string[] { BicepMediaTypes.BicepProviderArtifactLayerV1TarGzip, BicepMediaTypes.BicepProviderArtifactLayerV1TarGzip },
```

Comment:
why were these removed?

Reply:
because the restriction to have a single artifact layer is not part of the contract, I relaxed the contract to allow for these.

Reply:
This is where we get conflicts with module vs provider vs generic OCI.  What I expected was that the OCI code would split into generic, module and provider.  A single module layer should be part of the module contract, but not the generic or provider layers.

That's why I had GetSingleLayerOfMediaType available that the module code could use but provider doesn't need to.

Perhaps that's more picky than required.  What do you think?

Reply:
I apologize but I am struggling to understand. What I understand is that we would like separate implementations for Modules, Providers and a potential 'Other' type of artifact. I agree in principle but in practice this is quite challenging due to the way the `ModuleRegistry` class is built. 

To move forward I am simply moving the validation of the concrete artifact type to a constructor which I think adresses the concern. Please free to reopen otherwise.